client/src/app/components/registration/registration.component.ts

```diff
@@ -42,16 +44,19 @@ export class RegistrationComponent {
   }
 
   async submitForm() {
-    //TODO need to update and connect with server endpoint
-    console.log('Registered successfully');
-    console.log(this.registrationForm.value);
+    if (this.registrationForm.valid) {
+      this.userService.createUser(this.registrationForm.value).subscribe(() => {
+        this.router.navigate(['/login']);
+      });
+    }
   }
 }
 
 export const confirmPasswordValidator: ValidatorFn = (
   control: AbstractControl
 ): ValidationErrors | null => {
-  return control.value.password1 === control.value.password2
-    ? null
-    : { PasswordNoMatch: true };
+  const password = control.get('password')?.value;
+  const password2 = control.get('password2')?.value;
```

Comment:
Nice work using two password fields! This is a standard UI practice that's good to have